python/ray/util/ray_executor.py

```diff
@@ -0,0 +1,237 @@
+import itertools
+import time
+from concurrent.futures import Executor
+from functools import partial
+
+import ray
+from ray.util.annotations import PublicAPI
+
+
+@PublicAPI(stability="alpha")
+class RayExecutor(Executor):
+    """`RayExecutor` is a drop-in replacement for `ProcessPoolExecutor` and
+    `ThreadPoolExecutor` from `concurrent.futures` but distributes and executes
+    the specified tasks over a Ray cluster instead of multiple processes or
+    threads.
+    """
+
+    def __init__(self, max_workers=None, **kwargs):
+        """Initialises a new RayExecutor instance which distributes tasks over
+        a Ray cluster.
+
+        Args:
+            max_workers:
+                The maximum number of Ray workers to distribute the
+                tasks over. This is achieved by creating a fixed pool
```

Comment:
Maybe we need to change the docstring here? IIUC the behavior is

1. If max_workers=None, it is distributed to tasks (and the pool size is as big as num_cpus from the cluster)
2. otherwise, it creates actors using ray actor pool.

Also, I think we don't need to explain actor pool from the docstring (seems like an implementation detail, and I think we can implement this using tasks too)